gamestonk_terminal/economy/fred/fred_controller.py

```diff
@@ -0,0 +1,293 @@
+    CHOICES = ["cls", "?", "help", "q", "quit"]
+
+    CHOICES_COMMANDS = ["search", "add", "rmv", "plot"]
+
+    CHOICES += CHOICES_COMMANDS
+
+    def __init__(self):
+        """Constructor"""
+        self.fred_parser = argparse.ArgumentParser(add_help=False, prog="fred")
+        self.fred_parser.add_argument(
+            "cmd",
+            choices=self.CHOICES,
+        )
+        self.current_series = dict()
+
+    def print_help(self):
+        """Print help"""
+        id_string = ""
+        for s_id, title in self.current_series.items():
+            id_string += f"{s_id}: {title}\n"
+        help_text = f"""
+What do you want to do?
+    cls           clear screen
+    ?/help        show this menu again
+    q             quit this menu, and shows back to main menu
+    quit          quit to abandon program
+
+Current Series IDs:
+{id_string}
+    search        search FRED series notes
+    add           add series ID to list
+    rmv           remove series ID from list
+
+    plot          plot selected series
+        """
+        print(help_text)
+
+    def switch(self, an_input: str):
+        """Process and dispatch input
+
+        Returns
+        -------
+        MENU_GO_BACK, MENU_QUIT, MENU_RESET
+            MENU_GO_BACK - Show main context menu again
+            MENU_QUIT - Quit terminal
+            MENU_RESET - Reset terminal and go back to same previous menu
+        """
+
+        # Empty command
+        if not an_input:
+            print("")
+            return None
+
+        (known_args, other_args) = self.fred_parser.parse_known_args(an_input.split())
+
+        # Help menu again
+        if known_args.cmd == "?":
+            self.print_help()
+            return None
+
+        # Clear screen
+        if known_args.cmd == "cls":
+            system_clear()
+            return None
+
+        return getattr(
+            self, "call_" + known_args.cmd, lambda: "Command not recognized!"
+        )(other_args)
+
+    def call_help(self, _):
+        """Process Help command"""
+        self.print_help()
+
+    def call_q(self, _):
+        """Process Q command - quit the menu"""
+        return False
+
+    def call_quit(self, _):
+        """Process Quit command - exit the program"""
+        return True
+
+    @try_except
+    def call_search(self, other_args: List[str]):
+        """Process search command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="search",
+            description="Print series notes when searching for series. [Source: FRED]",
+        )
+        parser.add_argument(
+            "-s",
+            "--series",
+            action="store",
+            dest="series_term",
+            type=str,
+            required="-h" not in other_args,
+            help="Search for this series term.",
+        )
+        parser.add_argument(
+            "-n",
+            "--num",
```

Comment:
In the sia menu this arg is `-l` `--limit` in the crypto menu at first `-t` `--top` was proposed but then changed to `-l` like in the sia. Can we change it to `-l` to be consistent?

Reply:
So -n/--num has been used everywhere to date.  sia should be changed

Reply:
Yeah right, `-n` is used in more menus, although it is also used inconsistently. It can be `num` or `number`.

`--num` (either `--num` or `--number`) is used in
- crypto/nft, crypto/prediction
- economy
- etf
- forex/ba
- portfolio/brokers/ally, portfolio/po
- stocks, stocks/ba, stocks/dps, stocks/disc, stocks/dd, stocks/fmp, stocks/gov, stocks/insider, stocks/options, stocks/pred, stocks/qa

`--limit` is used in:
- crypto, crypto/defi, crypto/disc, crypto/dd,
- etf,
- forex/ba,
- portfolio/brokers/coinbase, portfolio/brokers/degiro,
- stocks/ba, stocks/ca, stocks/screener, stocks/sia

We were just discussing it with @DidierRLopes a couple of hours ago. We should decide on which one to use `--num`, `--number`, `--lim`, `--limit` and I can go and clean up the controllers across the terminal.

Reply:
I agree, and think number is a shit argument. I was the one who probably started using it, but it's not descriptive enough. Whereas limit you know you are limiting something!

Reply:
> I agree, and think number is a shit argument. I was the one who probably started using it, but it's not descriptive enough. Whereas limit you know you are limiting something!

can confirm.  u made me change it all in the early days :(